arangod/Replication2/ReplicatedLog/LogLeader.cpp

```diff
@@ -822,6 +832,41 @@ replicated_log::LogLeader::GuardedLeaderData::GuardedLeaderData(replicated_log::
                                                                 InMemoryLog inMemoryLog)
     : _self(self), _inMemoryLog(std::move(inMemoryLog)) {}
 
+auto replicated_log::LogLeader::release(LogIndex doneWithIdx) -> Result {
+  return _guardedLeaderData.doUnderLock([&](GuardedLeaderData& self) -> Result {
+    TRI_ASSERT(doneWithIdx <= self._inMemoryLog.getLastIndex());
+    if (doneWithIdx <= self._releaseIndex) {
+      return {};
+    }
+    self._releaseIndex = doneWithIdx;
+    LOG_CTX("a0c95", TRACE, _logContext) << "new release index set to " << self._releaseIndex;
+    return self.checkCompaction();
+  });
+}
+
+auto replicated_log::LogLeader::GuardedLeaderData::checkCompaction() -> Result {
+  auto const compactionStop = std::min(_largestCommonIndex, _releaseIndex);
```

Comment:
Didn't we want to change this to `_releaseIndex + 1`?

Reply:
We did on the follower side. I forgot the leader :headdesk:
```suggestion
  auto const compactionStop = std::min(_largestCommonIndex, _releaseIndex+1);
```